jobs/taar_locale.py

```diff
@@ -0,0 +1,449 @@
+repository at:
+
+ * https://github.com/mozilla/python_mozetl/blob/13266cf/mozetl/taar/taar_locale.py
+ * https://github.com/mozilla/python_mozetl/blob/13266cf/tests/test_taar_locale.py
+"""
+
+import boto3
+import click
+import json
+import logging
+import os
+
+from datetime import date, timedelta
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col
+from pyspark.sql.types import StructType, StructField, StringType, IntegerType
+from pandas import DataFrame, IndexSlice
+from numpy.random import laplace as rlaplace
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+LOCALE_FILE_NAME = "top10_dict"
+# Set the max multiplicative DP difference in probabilities to
+# exp(espilon) ~= 1.5.
+EPSILON = 0.4
+
+
+def get_client_addons(spark, start_date):
+    """Returns a Spark DF listing add-ons by client_id and locale.
+
+    Only Firefox release clients are considered. The query finds each client's
+    most recent record in the `clients_daily` dataset over the given time period
+    and returns its installed add-ons. System add-ons, disabled add-ons, and
+    unsigned add-ons are filtered out.
+
+    :param start_date: the earliest submission date to include (yyyymmdd)
+    :return: a DF with columns `locale`, `client_id`, `addon`
+    """
+    addons_query_template = """
+    WITH sample AS (
+        SELECT
+            client_id,
+            '{start_date}' as submission_date_s3,
+            locale,
+            active_addons
+        FROM clients_daily
+        WHERE
+            app_name='Firefox'
+            AND channel='release'
+            AND client_id IS NOT NULL
+    ),
+    sample_dedup AS (
+        SELECT
+            client_id,
+            locale,
+            explode(active_addons) AS addon_info
+        FROM (
+            SELECT
+                *,
+                -- retain the most recent active day for each client
+                row_number() OVER (
+                    PARTITION BY client_id
+                    ORDER BY submission_date_s3 DESC
+                ) AS idx
+            FROM sample
+        )
+        WHERE idx = 1
+    )
+    SELECT
+        locale,
+        client_id,
+        addon_info.addon_id as addon
+    FROM sample_dedup
+    WHERE
+        addon_info.blocklisted = FALSE -- not blocklisted
+        AND addon_info.type = 'extension' -- nice webextensions only
+        AND addon_info.signed_state = 2 -- fully reviewed addons only
+        AND addon_info.user_disabled = FALSE -- active addons only get counted
+        AND addon_info.app_disabled = FALSE -- exclude compatibility disabled addons
+        AND addon_info.is_system = FALSE -- exclude system addons
+        AND locale <> 'null'
+        AND addon_info.addon_id IS NOT NULL
+    """
+
+    addons_query = addons_query_template.format(start_date=start_date)
+
+    s3_url = "gs://moz-fx-data-derived-datasets-parquet/clients_daily/v6/submission_date_s3={}".format(
```

Comment:
This is no longer s3:
```suggestion
    gcs_uri = "gs://moz-fx-data-derived-datasets-parquet/clients_daily/v6/submission_date_s3={}".format(
```

Reply:
I've fixed this to be gs_url now